spec/coffee/directives/api/circle.spec.coffee

```diff
@@ -0,0 +1,129 @@
+describe 'uiGmapCircle', ->
+  allDone =  undefined
+  rootScope = null
+  timeout = null
+  GCircle = null
+  digest = null
+  modelClicked = false
+  browser = null
+
+  afterEach ->
+    digest = null
+    GCircle.resetInstances()
+    modelClicked = false
+
+  beforeEach ->
+    digest = (fn) =>
+     fn()
+     timeout?.flush() if browser.deferredFns?.length
+     rootScope?.$digest()
```

Comment:
The guards are there because at times other tests fail to totally irrelevant other specs. IE I have seen rootScope and timeout be null. Like you said I haven't had the time or desire to dig super deep into it.